Add option to enforce alignment constraint when planning memory

skrtskrtfb opened the pull request:
Summary: Add ability to enforce start alignment in Cadence Memory Planner

Differential Revision: D68762973

zonglinpeng commented on backends/cadence/aot/compiler.py:
```diff
@@ -257,6 +257,7 @@ def export_to_executorch_gen_etrecord(
     alloc_graph_output: bool = True,
     memory_config: Optional[MemoryConfig] = None,
     dump_graphs: bool = False,
+    mem_alignment: int = 0,
```
is this the default memory block size for alignment?

Commit:
Add option to enforce alignment constraint when planning memory (#8003)

Summary:

Add ability to enforce start alignment in Cadence Memory Planner

Reviewed By: zonglinpeng

Differential Revision: D68762973
